pkg/resourceruns/job/reconcile.go

```diff
@@ -175,3 +181,66 @@ func (r Reconciler) cleanPlanFiles(run *model.ResourceRun) {
 		}
 	})
 }
+
+// SetStatusFalse sets the status of the resource run to false with k8s job.
+func setStatusFalse(run *model.ResourceRun, job *batchv1.Job, errMsg string) {
+	if status.ResourceRunStatusPending.IsUnknown(run) {
+		errMsg = fmt.Sprintf("pending failed: %s", errMsg)
+		status.ResourceRunStatusPending.False(run, errMsg)
+	}
+
+	var (
+		taskTypeFromK8s = taskTypeFromK8sJob(job)
+		ct              status.ConditionType
+	)
+	switch taskTypeFromK8s {
+	case types.RunTaskTypePlan:
+		ct = status.ResourceRunStatusPlanned
+		errMsg = fmt.Sprintf("plan failed: %s", errMsg)
+	case types.RunTaskTypeApply, types.RunTaskTypeDestroy:
+		ct = status.ResourceRunStatusApplied
+		errMsg = fmt.Sprintf("apply failed: %s", errMsg)
+	}
+
+	if ct.IsUnknown(run) {
+		ct.False(run, errMsg)
+	}
+
+	run.Status.SetSummary(status.WalkResourceRun(&run.Status))
+}
+
+// setStatusTrue sets the status of the resource run to true with k8s jobs.
+func setStatusTrue(run *model.ResourceRun, job *batchv1.Job, msg string) {
+	var (
+		taskTypeFromK8s = taskTypeFromK8sJob(job)
+		ct              status.ConditionType
+	)
+
+	switch taskTypeFromK8s {
+	case types.RunTaskTypePlan:
+		ct = status.ResourceRunStatusPlanned
+		msg = ""
+	case types.RunTaskTypeApply, types.RunTaskTypeDestroy:
+		ct = status.ResourceRunStatusApplied
+	}
+
+	if ct.IsUnknown(run) {
+		ct.True(run, msg)
+	}
+
+	run.Status.SetSummary(status.WalkResourceRun(&run.Status))
+}
+
+// taskTypeFromK8sJob returns the task type from k8s job.
+func taskTypeFromK8sJob(job *batchv1.Job) types.RunJobType {
+	switch {
+	case strings.Contains(job.Name, string(types.RunTaskTypeApply)):
+		return types.RunTaskTypeApply
+	case strings.Contains(job.Name, string(types.RunTaskTypePlan)):
+		return types.RunTaskTypePlan
+	case strings.Contains(job.Name, string(types.RunTaskTypeDestroy)):
+		return types.RunTaskTypeDestroy
+	default:
+		return "unknown"
```

Comment:
Do we have labels to differentiate the task type?

Reply:
Now we don't have, but we can add.